### PR TITLE
Windows fix

### PR DESCRIPTION
**Pull request by thxCode**
This PR is going to do as below:
- Change the proxy mode for running Windows kube-proxy.exe from `userspace` to `kernelspace`, fix https://github.com/rancher/rancher/issues/15510
- Adjust the Windows part on worker node to ignore log-linker running
- Patch https://github.com/rancher/rancher/pull/13796#discussion_r212535628
- Reflact the Powershell scripts:
   - take out the HNSNetwork operation into a sperate file: hns.psm1
   - add procces checking when each components starting
   - fix a configuration mistake of Flannel net-config.json

### CODE REVIEW

**Comment by thxCode** [comment]
https://github.com/rancher/rancher/issues/15510

File: pkg/rkeworker/docker.go [hunk 1]
```diff
@@ -280,6 +286,10 @@ func natPortSetToSlice(args map[nat.Port]struct{}) []nat.Port {
 }
 
 func runLogLinker(ctx context.Context, c *client.Client, containerName string, p v3.Process) error {
+	if ignoreWindows(ctx) {
```
Comment (StrongMonkey) [inline pkg/rkeworker/docker.go]: I feel like this can be detected using runtime.GOOS.

Comment (thxCode) [reply pkg/rkeworker/docker.go]: Ok, got

**Comment by StrongMonkey** [comment]
LGTM. @thxCode Can you squash your commits into a single one?

File: pkg/rkeworker/docker.go [hunk 2]
```diff
@@ -201,17 +202,23 @@ func changed(ctx context.Context, c *client.Client, expectedProcess v3.Process,
...
 				es := strings.SplitN(env, "=", 2)
-				expectedEnvs[es[0]] = es[1]
+				if len(es) == 2 {
+					expectedEnvs[es[0]] = es[1]
+				}
 			}
 			for _, env := range expectedProcess.Env {
 				es := strings.SplitN(env, "=", 2)
-				expectedEnvs[es[0]] = es[1]
+				if len(es) == 2 {
```
Comment (cjellick) [inline pkg/rkeworker/docker.go]: this is a valid env var config in a docker inspect:
```
"Env": [
                "foo",
                "foo=bar",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
```
When the env var is just a key without a value, it tells docker to inherit the value from the host. Do you want to account for that scenario?

Comment (thxCode) [reply pkg/rkeworker/docker.go]: @cjellick, yes

Comment (cjellick) [reply pkg/rkeworker/docker.go]: It will be in the container with the value that that variable had on the host

**Comment by cjellick** [comment]
Per @StrongMonkey's previous comment, please squash your commits to 1 (or more than one if they need to be logically separated) commit with meaningful messages.
For example, there should not be an "address comments" commit

File: pkg/rkeworker/docker.go (hunk 2 above)
Comment (thxCode) [reply pkg/rkeworker/docker.go]: Oh, I got that. You means `foo1` always a field name in `expectedProcess.Env`, but maybe have field value in `actualProcess.Env`.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**